app/src/androidTest/java/com/omelan/cofi/pages/settings/licenses/TimerSettingsTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.omelan.cofi.pages.settings.licenses
+
+import android.content.Context
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createComposeRule
+import com.omelan.cofi.LocalPiPState
+import com.omelan.cofi.pages.settings.TimerSettings
+import com.omelan.cofi.ui.CofiTheme
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class TimerSettingsTest {
+    @get:Rule
+    val composeTestRule = createComposeRule()
+    private lateinit var context: Context
+
+    private fun renderTimerSettings() {
+        composeTestRule.setContent {
+            context = LocalContext.current
+            CofiTheme {
+                CompositionLocalProvider(
+                    LocalPiPState provides false,
+                ) {
+                    TimerSettings(goBack = { })
+                }
+            }
+        }
+    }
+
+    @Test
+    fun showTimerSettings() {
+        composeTestRule.onNodeWithTag("settings_timer_list_item_pip").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("settings_timer_list_item_sound").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("settings_timer_list_item_vibration").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("settings_timer_list_item_weight").assertIsDisplayed()
+    }
+
+    @Test
+    fun checkPipSettings() {
+        renderTimerSettings()
+        composeTestRule.onNodeWithTag("settings_timer_list_item_pip")
+            .assertIsDisplayed()
+        composeTestRule.onNodeWithTag("settings_timer_switch_pip")
+            .assertIsOn().performClick().assertIsOff()
```

Comment:
That's fine, but how about something more thorough. Right now you are checking only clicking on the `Switch`, but user can also press `ListItem` to change this value. So maybe add a test that checks if `Switch` state (if it's on or off) changes after pressing on a `ListItem`

I would also suggest moving it to separate function like 
```kt
    fun checkListItem(listItemTag: String, switchTag: String) {
        val listItem = composeTestRule.onNodeWithTag(listItemTag)
        val switch = composeTestRule.onNodeWithTag(switchTag)
        
        listItem.assertIsDisplayed() 
        […] // plus that click test I've talked about above
        switch.assertIsOn().performClick().assertIsOff()
        […]
    }
```
That way instead of changing all three tests you would have only one place to change

Reply:
Hey, thanks so much for your suggestions 😃 I'm posting an update based on your advice